### PR TITLE
Make asynchronous Slack tests run properly

### LINKED ISSUE #4664: Some async tests are not properly run
5 asynchronous test methods are currently never awaited, so the tests pass pytest regardless of if the method itself should actually pass. Example: https://travis-ci.com/RasaHQ/rasa/jobs/248702574#L4028

In the case of slack, this might have been introduced during the change from `httpretty` to `responses`.

### PR DESCRIPTION
**Pull request by alwx**
This PR introduces a new dependency (`pytest-asyncio`) because a suitable async framework is required to run asynchronous tests. `pytest-sanic` doesn't work in this particular situation — it's designed only to help testing sanic application with async coroutines.

**Proposed changes**:
- fix #4664

**Status (please check what you already did)**:
- [ ] fixed tests
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/core/test_channels.py [hunk 1]
```diff
@@ -586,165 +587,179 @@ def test_slackbot_init_two_parameter():
...
+            "channel": ["General"],
+            "as_user": ["True"],
+            "attachments": [json.dumps([attachment])],
+        }
 
 
 @pytest.mark.filterwarnings("ignore:unclosed.*:ResourceWarning")
+@pytest.mark.asyncio
 @responses.activate
 async def test_slackbot_send_attachment_withtext():
```
Comment (ricwo) [inline tests/core/test_channels.py]:
```suggestion
async def test_slackbot_send_attachment_with_text():
```

File: tests/core/conftest.py [hunk 2]
```diff
@@ -84,6 +84,13 @@ def __init__(
         super(MongoTrackerStore, self).__init__(_domain, None)
 
 
+@pytest.yield_fixture(scope="session")
+def event_loop(request):
```
Comment (ricwo) [inline tests/core/conftest.py]: why is this fixture necessary? it's not passed as an argument into any test function

File: tests/nlu/base/test_evaluation.py [hunk 3]
```diff
@@ -52,6 +52,13 @@
 from rasa.nlu.test import is_response_selector_present
 
 
+@pytest.yield_fixture(scope="session")
+def event_loop(request):
+    loop = asyncio.get_event_loop_policy().new_event_loop()
+    yield loop
+    loop.close()
+
+
```
Comment (ricwo) [inline tests/nlu/base/test_evaluation.py]: why is this needed here again? From what I understood, it's only used if you pass the `event_loop` fixture into a test function, or the `@pytest.mark.asyncio` decorator is used

File: tests/core/test_channels.py (hunk 1 above)
Comment (alwx) [reply tests/core/test_channels.py]: 👍

File: tests/core/conftest.py (hunk 2 above)
Comment (alwx) [reply tests/core/conftest.py]: The reason why it's needed described here: https://github.com/pytest-dev/pytest-asyncio/issues/68

Long story short: this `event_loop` is used by `pytest-asyncio`, and redefining it is currently the only supported way of changing the scope of this fixture. So basically the only line that makes real sense here is this one:
```
@pytest.yield_fixture(scope="session")
```

Changing the scope of this function is required to avoid ScopeMismatch errors like this one:
```
ScopeMismatch: You tried to access the 'function' scoped fixture 'event_loop' with a 'session' scoped request object, involved factories
```

Comment (alwx) [reply tests/core/conftest.py]: I should actually comment it there.

File: tests/nlu/base/test_evaluation.py (hunk 3 above)
Comment (alwx) [reply tests/nlu/base/test_evaluation.py]: We have to redefine it because `tests/core/conftest.py` is not imported to `tests/nlu/`
